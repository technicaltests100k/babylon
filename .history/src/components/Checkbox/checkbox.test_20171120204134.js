import React, { Component } from 'react';

class Checkbox extends Component {
  render() {
    return <div />;
  }
}

export default Checkbox;
