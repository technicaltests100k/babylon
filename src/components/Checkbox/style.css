.checkbox {
  width: 20px;
  height: 20px;
  border-style: inset;
  display: inline-block;
  vertical-align: bottom;
}

.checkbox:hover {
  cursor: pointer;
}

.checkboxIcon {
  width: 20px;
  height: 20px;
  fill: green;
}
