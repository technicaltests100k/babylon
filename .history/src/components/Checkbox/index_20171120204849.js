import React, { Component } from 'react';
import style from './style.css';

class Checkbox extends Component {
  render() {
    return <div className={style} />;
  }
}

export default Checkbox;
