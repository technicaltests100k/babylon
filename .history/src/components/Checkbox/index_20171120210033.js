import React, { Component } from 'react';
import './style.css';

class Checkbox extends Component {
  constructor() {
    super(props);
    this.state = {
      isChecked: false
    };
  }

  handleOnClick = () => {
    this.setState({
      isChecked: !this.state.isChecked
    });
  };

  render() {
    return <div className={'checkbox'} onClick={this.handleOnClick} />;
  }
}

export default Checkbox;
