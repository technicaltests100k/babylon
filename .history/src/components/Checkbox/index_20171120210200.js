import React, { Component } from 'react';
import './style.css';

class Checkbox extends Component {
  constructor() {
    this.state = {
      isChecked: false
    };
  }

  handleOnClick = () => {
    this.setState({
      isChecked: !this.state.isChecked
    });
  };

  render() {
    return <div className={'checkbox'} onClick={this.handleOnClick} />;
  }
}

export default Checkbox;
