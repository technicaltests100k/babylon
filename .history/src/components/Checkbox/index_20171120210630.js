import React, { Component } from 'react';
import './style.css';

class Checkbox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isChecked: false
    };
  }

  handleOnClick = () => {
    console.log('state');
    this.setState({
      isChecked: !this.state.isChecked
    });
  };

  render() {
    const style = this.isChecked ? 'checked' : '';
    return <div className={`checkbox ${style}`} onClick={this.handleOnClick} />;
  }
}

export default Checkbox;
