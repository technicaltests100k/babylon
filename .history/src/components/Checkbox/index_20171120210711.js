import React, { Component } from 'react';
import './style.css';

class Checkbox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isChecked: false
    };
  }

  handleOnClick = () => {
    this.setState({
      isChecked: !this.state.isChecked
    });
  };

  render() {
    const style = this.state.isChecked ? 'checked' : '';
    return <div className={`checkbox ${style}`} onClick={this.handleOnClick} />;
  }
}

export default Checkbox;
