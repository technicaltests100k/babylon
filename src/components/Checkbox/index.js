import React, { Component } from 'react';
import './style.css';
import icon from '../../icons.svg';

class Checkbox extends Component {
  constructor() {
    super();
    this.state = {
      isChecked: false
    };
  }

  handleOnClick = () => {
    this.setState({
      isChecked: !this.state.isChecked
    });
  };

  render() {
    const style = this.state.isChecked ? 'checked' : '';
    return (
      <div className={`checkbox ${style}`} onClick={this.handleOnClick}>
        <svg className={'icon'}>
          <use xlinkHref={`${icon}#icon-checkmark`} />
        </svg>
      </div>
    );
  }
}

export default Checkbox;
