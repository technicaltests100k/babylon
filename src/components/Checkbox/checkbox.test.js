import React from 'react';
import Checkbox from './';

/**
* Test for <Checkbox /> component.
* @test {Checkbox}
*/
const props = {
  checked: false,
  extendStyle: 'classToExtend'
};

describe('<Checkbox />', () => {
  it('should render Checkbox with all the correct props and no icon', () => {
    const wrapper = render(<Checkbox {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Checkbox and show the icon', () => {
    const wrapper = render(<Checkbox {...props} checked />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should update the state when clicked multiple times', () => {
    const wrapper = shallow(<Checkbox {...props} />);

    expect(wrapper.find('svg').exists()).toEqual(false);

    wrapper.simulate('click');

    expect(wrapper.find('svg').exists()).toEqual(true);

    wrapper.simulate('click');

    expect(wrapper.find('svg').exists()).toEqual(false);
  });
});
