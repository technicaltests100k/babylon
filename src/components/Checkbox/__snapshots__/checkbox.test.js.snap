// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render Checkbox and show the icon 1`] = `
<div
  class="checkbox classToExtend"
>
  <svg
    class="checkboxIcon"
  >
    <use
      href="icons.svg#icon-checkmark"
    >
      
    </use>
  </svg>
</div>
`;

exports[`<Checkbox /> should render Checkbox with all the correct props and no icon 1`] = `
<div
  class="checkbox classToExtend"
>
  
</div>
`;
